src/main/java/io/kestra/plugin/pulsar/BaseProducer.java

```diff
@@ -0,0 +1,199 @@
+package io.kestra.plugin.pulsar;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.pulsar.client.api.*;
+
+import io.kestra.core.models.executions.metrics.Counter;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+
+public abstract class BaseProducer<T> {
```

Comment:
```suggestion
public abstract class AbstractProducer<T> {
```

By convention, an abstract class is prefixed by Abstract.